plugins/utils/spectre.nix

```diff
@@ -0,0 +1,246 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib; {
+  options.plugins.spectre = let
+    option = defaults:
+      helpers.defaultNullOpts.mkNullable
+      (with types; attrsOf attrs)
+      defaults
+      "placeholder option";
+  in
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "nvim-spectre";
+
+      package = helpers.mkPackageOption "nvim-spectre" pkgs.vimPlugins.nvim-spectre;
+
+      colorDevicon = helpers.defaultNullOpts.mkBool true "If set to true will color devicons";
+
+      openCmd = helpers.defaultNullOpts.mkStr "vnew" "Command to run when opening spectre";
+
+      liveUpdate = helpers.defaultNullOpts.mkBool false "If set to true will auto execute search again when you write to any file in vim";
+
+      lineSepStart = helpers.defaultNullOpts.mkStr "┌-----------------------------------------" "Shown to Separate search and search results";
+
+      resultPadding = helpers.defaultNullOpts.mkStr "¦  " "Shown for padding the result on the left";
+
+      lineSep = helpers.defaultNullOpts.mkStr "└-----------------------------------------" "Shown at the bottom of the search results";
+
+      highlight = {
+        ui = helpers.defaultNullOpts.mkStr "String";
+        search = helpers.defaultNullOpts.mkStr "DiffChange";
+        replace = helpers.defaultNullOpts.mkStr "DiffDelete";
+      };
+
+      mappings =
+        option
```

Comment:
Im not really sure what to type this tbh, i created a type above but am not confident its close to being correct.